### DISCUSSION TITLE
How to enable Location access

### DISCUSSION THREAD

**Discussion by schandra09net** [cefsharp/CefSharp#3719] [General]
Hello Team,

1. I am currently using Cef 91.1.230 In my application development with C# and WPF.
2. Tried loading a page in my CefBrowser with link "https://earth.google.com/web/@0,0,0a,22251752.77375655d,35y,0h,0t,0r" and clicked on Location icon on bottom right corner.
3. Got an warning that "Location servies should be enabled"

Possible solutions tried:
1. Followed couple of articles to use IGeoLocationHandler but unfortunately which is not there/removed in v91.
2. Tried using "ViewControl.WebBrowser.GetDevToolsClient().Browser.GrantPermissionsAsync(new PermissionType[] { PermissionType.Geolocation });" but getting an error "CefSharp.DevTools.DevToolsClientException: {"code":-32000,"message":"Browser context management is not supported."}'

How to enable the "Location services", can any one help with this.

**Comment by amaitland** [comment]
Geolocation is not supported. Upstream support was removed.

**Comment by campersau** [comment]
Another option is to mock the relevant geolocation APIs in JavaScript yourself like mentioned in https://magpcss.org/ceforum/viewtopic.php?f=6&t=17432

Here is an example, Google Earth uses `watchPosition` whereas Google Maps uses `getCurrentPosition`:
```c#
browser.ExecuteScriptAsyncWhenPageLoaded(@"
navigator.permissions.query = options => {
  return Promise.resolve({
    state: 'granted'
  });
};
navigator.geolocation.getCurrentPosition = (success, error, options) => {
  success({
    coords: {
      latitude: -33.854477,
      longitude: 151.234738,
      accuracy: 10,
      altitude: null,
      altitudeAccuracy: null,
      heading: null,
      speed: null
    },
    timestamp: Date.now()
  });
};
navigator.geolocation.watchPosition = (success, error, options) => {
  success({
    coords: {
      latitude: -33.854477,
      longitude: 151.234738,
      accuracy: 49,
      altitude: null,
      altitudeAccuracy: null,
      heading: null,
      speed: null
    },
    timestamp: Date.now()
  });
  return 0;
};
", oneTime: false);
```

See https://developer.mozilla.org/docs/Web/API/Geolocation for the JavaScript APIs..

Cef references to removed geolocation APIs:
https://bitbucket.org/chromiumembedded/cef/commits/ac86b61
https://bitbucket.org/chromiumembedded/cef/issues/2380/geolocation-can-not-work-after-branch-3239

**Reply by schandra09net** [reply]
@campersau : Thank you so much again, now it navigate to fixed custom location as specified in Longitude and Latitude after added the script to the browser in IsBrowserInitializedChanged event.

Thanks Amigo.
